sdk/src/main/java/com/hedera/hashgraph/sdk/ManagedNode.java

```diff
@@ -228,13 +237,48 @@ synchronized ManagedChannel getChannel() {
         }
 
         channel = channelBuilder
+            .keepAliveTimeout(10, TimeUnit.SECONDS)
             .userAgent(getUserAgent())
             .executor(executor)
             .build();
 
         return channel;
     }
 
+    boolean channelFailedToConnect() {
+        if (hasConnected) {
+            return false;
+        }
+        @Var var state = getChannel().getState(true);
+        try {
+            for (@Var int i = 0; i < GET_STATE_MAX_ATTEMPTS && state != ConnectivityState.READY; i++) {
```

Comment:
Is this one of those rare case where a do-while loop would work?

Reply:
Needs to keep track of number of attempts, doesn't it?